src/storage-client/src/metrics.rs

```diff
@@ -151,6 +191,21 @@ pub struct ReplicaMetrics {
     inner: Arc<ReplicaMetricsInner>,
 }
 
+impl ReplicaMetrics {
+    /// Observe a successful replica connection.
+    pub fn observe_connect(&self, wait_time: Duration) {
+        self.inner.replica_connects_total.inc();
+        self.observe_connect_time(wait_time);
+    }
+
+    /// Observe time spent waiting for a replica connection.
+    pub fn observe_connect_time(&self, wait_time: Duration) {
+        self.inner
+            .replica_connect_wait_time_seconds_total
+            .inc_by(wait_time.as_secs_f64());
+    }
```

Comment:
Nit: From the name alone, I wouldn't have `observe_connect` expected to accept a `wait_time` parameter. Consider calling `observe_connect_time` explicitly instead of threading it through `observe_connect`.

Reply:
Done. I also restructured the two `ReplicaTask::connect` to make them more similar.